sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SpannerMutationCoder.java

```diff
@@ -0,0 +1,224 @@
+package org.apache.beam.sdk.coders;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Date;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Timestamp;
+import com.google.cloud.spanner.Type;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.spanner.ValueBinder;
+import com.google.common.io.ByteArrayDataInput;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+/**
+ * A {@link Coder} that encodes Spanner {@link Mutation} objects.
+ */
+public class SpannerMutationCoder extends AtomicCoder<Mutation> {
```

Comment:
Mutations are eventually converted to protobufs, we should use proto serialization here. Otherwise, this class has to be tested extensively.

I'm looking at 
https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-spanner/src/main/java/com/google/cloud/spanner/Mutation.java#L337 and toProto method has package visibility. We need either expose to/from proto methods in the SDK, make Mutation serializable or replicate that code here.